eva-server/src/main/java/uk/ac/ebi/eva/server/repository/VariantEntityRepositoryImpl.java

```diff
@@ -40,31 +44,108 @@
+                                                             Double siftValue, Pageable pageable) {
         Query query = new Query(Criteria
                                         .where("chr").is(chr)
                                         .and("start").lte(end).gt(start - MARGIN)
                                         .and("end").gte(start).lt(end + MARGIN)
         );
 
+        if (consequenceType != null && !consequenceType.isEmpty()) {
+            queryConsequenceType(query, consequenceType);
+        }
+
+        if (mafValue != null && mafOperator != VariantEntityRepository.RelationalOperator.NONE) {
+            queryMaf(query, mafValue, mafOperator);
+        }
+
+        if (polyphenScoreValue != null && polyphenScoreOperator != VariantEntityRepository.RelationalOperator.NONE) {
+            queryPolyphenScore(query, polyphenScoreValue, polyphenScoreOperator);
+        }
+
+        if (siftValue != null && siftOperator != VariantEntityRepository.RelationalOperator.NONE) {
+            querySift(query, siftValue, siftOperator);
+        }
+
+        if (studies != null && !studies.isEmpty()) {
+            queryStudies(query, studies);
+        }
+
         ArrayList<String> sortProps = new ArrayList<String>();
         sortProps.add("chr");
         sortProps.add("start");
         query.with(new Sort(Sort.Direction.ASC, sortProps));
 
+        Pageable pageable1 = (pageable != null) ? pageable : new PageRequest(0, 10);
+        query.with(pageable1);
+
         return mongoTemplate.find(query, VariantEntity.class);
     }
+
+    void queryConsequenceType(Query query, List<String> consequenceType) {
+        List<Integer> consequenceTypeConv = consequenceType.stream()
+                                                           .map(c -> Integer.parseInt(c.replaceAll("[^\\d.]", ""), 10))
+                                                           .collect(Collectors.toList());
+        query.addCriteria(Criteria.where("annot.ct.so").in(consequenceTypeConv));
+    }
+
+    void queryMaf(Query query, double mafValue, VariantEntityRepository.RelationalOperator mafOperator) {
+        relationalCriteriaHelper(query, "st.maf", mafValue, mafOperator);
+    }
+
+    void queryPolyphenScore(Query query, double polyphenScoreValue,
+                            VariantEntityRepository.RelationalOperator polyphenScoreOperator) {
+        relationalCriteriaHelper(query, "annot.ct.polyphen.sc", polyphenScoreValue, polyphenScoreOperator);
+    }
+
+    void querySift(Query query, double siftValue, VariantEntityRepository.RelationalOperator siftOperator) {
+        relationalCriteriaHelper(query, "annot.ct.sift.sc", siftValue, siftOperator);
+    }
+
+    void queryStudies(Query query, List<String> studies) {
+        query.addCriteria(Criteria.where("files.sid").in(studies));
+    }
+
+    void relationalCriteriaHelper(Query query, String jsonPath, double value,
```

Comment:
A more meaningful name for the second argument could be `field`, given it represents which field will be checked. The fact that Mongo uses BSON documents is not really relevant.